pkg/resources/grant_ownership.go

```diff
@@ -0,0 +1,668 @@
+				},
+				"all": {
+					Type:        schema.TypeList,
+					Optional:    true,
+					ForceNew:    true,
+					Description: "Configures the privilege to be granted on all objects in either a database or schema.",
+					MaxItems:    1,
+					Elem: &schema.Resource{
+						Schema: grantOwnershipBulkOperationSchema("all"),
+					},
+					ExactlyOneOf: []string{
+						"on.0.object_name",
+						"on.0.all",
+						"on.0.future",
+					},
+				},
+				"future": {
+					Type:        schema.TypeList,
+					Optional:    true,
+					ForceNew:    true,
+					Description: "Configures the privilege to be granted on all objects in either a database or schema.",
+					MaxItems:    1,
+					Elem: &schema.Resource{
+						Schema: grantOwnershipBulkOperationSchema("future"),
+					},
+					ExactlyOneOf: []string{
+						"on.0.object_name",
+						"on.0.all",
+						"on.0.future",
+					},
+				},
+			},
+		},
+	},
+}
+
+func grantOwnershipBulkOperationSchema(branchName string) map[string]*schema.Schema {
+	return map[string]*schema.Schema{
+		"object_type_plural": {
+			Type:         schema.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			Description:  "Specifies the type of object in plural form on which you are transferring ownership. Available values are: AGGREGATION POLICIES | ALERTS | AUTHENTICATION POLICIES | COMPUTE POOLS | DATABASES | DATABASE ROLES | DYNAMIC TABLES | EVENT TABLES | EXTERNAL TABLES | EXTERNAL VOLUMES | FAILOVER GROUPS | FILE FORMATS | FUNCTIONS | HYBRID TABLES | ICEBERG TABLES | IMAGE REPOSITORIES | INTEGRATIONS | MATERIALIZED VIEWS | NETWORK POLICIES | NETWORK RULES | PACKAGES POLICIES | PIPES | PROCEDURES | MASKING POLICIES | PASSWORD POLICIES | PROJECTION POLICIES | REPLICATION GROUPS | ROLES | ROW ACCESS POLICIES | SCHEMAS | SESSION POLICIES | SECRETS | SEQUENCES | STAGES | STREAMS | TABLES | TAGS | TASKS | USERS | VIEWS | WAREHOUSES",
+			ValidateFunc: validation.StringInSlice(validGrantOwnershipPluralObjectTypesString, true),
+		},
+		"in_database": {
+			Type:             schema.TypeString,
+			Optional:         true,
+			ForceNew:         true,
+			Description:      "The fully qualified name of the database.",
+			ValidateDiagFunc: IsValidIdentifier[sdk.AccountObjectIdentifier](),
+			ExactlyOneOf: []string{
+				fmt.Sprintf("on.0.%s.0.in_database", branchName),
+				fmt.Sprintf("on.0.%s.0.in_schema", branchName),
+			},
+		},
+		"in_schema": {
+			Type:             schema.TypeString,
+			Optional:         true,
+			ForceNew:         true,
+			Description:      "The fully qualified name of the schema.",
+			ValidateDiagFunc: IsValidIdentifier[sdk.DatabaseObjectIdentifier](),
+			ExactlyOneOf: []string{
+				fmt.Sprintf("on.0.%s.0.in_database", branchName),
+				fmt.Sprintf("on.0.%s.0.in_schema", branchName),
+			},
+		},
+	}
+}
+
+func GrantOwnership() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: CreateGrantOwnership,
+		DeleteContext: DeleteGrantOwnership,
+		ReadContext:   ReadGrantOwnership,
```

Comment:
let's leave a comment that update is not needed because everything is forceNew. Also, should we have (is there already in the SDK) a no-op function we can just reference in such cases?

Reply:
I will try to set it to the NoOp function, but some time ago I had with difficulties with it, so nil != NoOp. Regardless, I'll add a note as to why there's no update.

Reply:
Noop cannot be set, because all of the fields are marked as ForceNew. I just left the comment why there's no Update.